packages/lesswrong/components/users/UsersSingle.jsx

```diff
@@ -1,10 +1,18 @@
-import { Components, registerComponent } from 'meteor/vulcan:core';
+import { Components, registerComponent, Utils } from 'meteor/vulcan:core';
 import React from 'react';
+import { withRouter } from 'react-router';
 
-const UsersSingle = (props, context) => {
-  return <Components.UsersProfile userId={props.params._id} slug={props.params.slug} />
+const UsersSingle = ({params, router}) => {
+  const slug = Utils.slugify(params.slug);
+  const canonicalUrl = `/users/${slug}`;
```

Comment:
Can you use `Users.getProfileUrl` here instead? Single source of truth for URL structure seems important.

Reply:
I suppose. I don't really like the source of truth living inside Vulcan rather than inside our code, though; it's probably too late to go back on it now, but it felt like a bug when the URLs switched from `/user/username` to `/users/username`.

Reply:
I guess you run into the same problem with the thing being typed to get a user, but us only having the slug. But seems worth the cost to refactor it.

Reply:
Actually no, there's a bigger issue: `Users.getProfileUrl` depends on having a user object (rather than just a slug), and there's no analogous version which uses only a slug. Fixing that would require synchronizing with a PR against the Vulcan side, which doesn't seem worth it.

Reply:
We could just overwrite it.